saleor/core/hashers.py

```diff
@@ -0,0 +1,24 @@
+import base64
+import hashlib
+from django.contrib.auth.hashers import PBKDF2PasswordHasher
+from django.utils.encoding import force_bytes
+
+
+class SHA512Base64PBKDF2PasswordHasher(PBKDF2PasswordHasher):
+    """
+        Symfony 5000 iterations of sha512 plus base64 encoder, with an extra PBKDF2 round.
+    """
+    algorithm = "sha512b64pbkdf2"
+    iterations = 690000
+
+    def encode(self, password, salt, iterations=None):
```

Comment:
Would be good to add a small test checking the logic from encode so that in case of a future refactor we don't break the logic here.

Reply:
It's already been tested.
I decided to keep the pbkdf2 round in a different method to be able to modify the original tokens.

```
# migration
original_token = "GYZwGbUT4VFqNgejNlAabTyI0UAyK1nWm7Bs5ME5K9lIn47s4D43SucLFGor3DRnolbnI9QD7zZerO7hhLyCpQ=="
migrated_token = SHA512Base64PBKDF2PasswordHasher().PBKDF2_round(
    password=original_token,
    salt=SHA512Base64PBKDF2PasswordHasher().salt()
)
assert SHA512Base64PBKDF2PasswordHasher().verify(
    "Breitling123!",
    migrated_token
)

# new password 
generated_token = SHA512Base64PBKDF2PasswordHasher().encode(
    "Breitling123!", salt=SHA512Base64PBKDF2PasswordHasher().salt()
)
assert 90 == len(generated_token)
assert SHA512Base64PBKDF2PasswordHasher().verify(
    "Breitling123!",
    generated_token
)
```

Reply:
@Dieg0x17 We need a test in code, not in comments. Something that guarantees this code still works in six months.

Reply:
There you have your pytest implementation. Thanks @aniav and @patrys